### ISSUE TITLE
`.pipe` signature change

### ISSUE THREAD

**Issue by hfhchan-plb** [google/zx#949]
Previously in 8.1.9, calling `.pipe(process.stdout)` would return ProcessPromise, now (8.2.2) it returns `D & PromiseLike<D>`. Is there any way to get the old behaviour?

**Comment by visnaut** [comment]
> Looks like this broke as part of [#921](https://github.com/google/zx/pull/921) and [#928](https://github.com/google/zx/pull/928), where the return type of `.pipe()` became polymorphic. Shouldn't this have been a major version change instead of minor?

Agreed. I have a script in CI that broke because of this change. 

The docs do not make reference to this change, and I believe [the third example in the `pipe` doc](https://google.github.io/zx/process-promise#pipe) won't work anymore, since I don't think you can chain them anymore.

**Comment by antongolub** [comment]
@visnaut

> The docs do not make reference to this change, and I believe [the third example in the pipe doc](https://google.github.io/zx/process-promise#pipe) won't work anymore, since I don't think you can chain them anymore.

We'll update the example: 
```
$`printf "hello"`
  .pipe($`awk '{printf $1", world!"}'`)
  .pipe($`tr '[a-z]' '[A-Z]'`)

// →
$`printf "hello"`
  .pipe`awk '{printf $1", world!"}'`
  .pipe`tr '[a-z]' '[A-Z]'`
```

**Comment by visnaut** [comment]
> We'll update the example:
> 
> $`printf "hello"`
>   .pipe($`awk '{printf $1", world!"}'`)
>   .pipe($`tr '[a-z]' '[A-Z]'`)
> 
> // →
> $`printf "hello"`
>   .pipe`awk '{printf $1", world!"}'`
>   .pipe`tr '[a-z]' '[A-Z]'`

Thanks, I appreciate the reply. 

Let me better articulate where I think the issue lies here: previously, I was doing the following:

```javascript
const lines = await $(options)`command`.pipe(process.stdout).lines();
```

In other words, taking advantage of the fact that any method defined for ProcessPromise would return itself, allowing for the above chaining.

I've since updated the above to something like so:
```javascript
const cli = $(options)`command`;

cli.pipe(process.stdout);

const lines = await cli.lines();
```

But you can see how anyone consuming the library and relying on the previous method signature to chain methods like that would be caught off guard by 8.2; hence why I agree with @hfhchan-plb that this maybe should have been a major version change.

**Comment by antongolub** [comment]
The problem is even broader, but I think it can be fixed. Stay tuned.

**Comment by antongolub** [comment]
@visnaut, @hfhchan-plb 

Could you plz check if [v8.2.2-dev.3b9afe1](https://www.npmjs.com/package/zx/v/8.2.2-dev.3b9afe1) solves your issue.

**Comment by antongolub** [comment]
Backward compatibility patch released as [v8.2.3](https://github.com/google/zx/releases/tag/8.2.3)
